0.data_analysis/plate_eda_analyses/nbconverted/single_cell_fs_plate_cell_count_eda.py

```diff
@@ -0,0 +1,394 @@
+cwd = pathlib.Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+
+# ## Define paths
+
+# ### Input paths
+
+# In[3]:
+
+
+data_path = pathlib.Path(root_dir / "nf1_painting_repo/3.processing_features/data/single_cell_profiles").resolve(strict=True)
+
+plate3df_path = pathlib.Path(data_path / "Plate_3_sc_feature_selected.parquet").resolve(strict=True)
+plate3pdf_path = pathlib.Path(data_path / "Plate_3_prime_sc_feature_selected.parquet").resolve(strict=True)
+plate5df_path = pathlib.Path(data_path / "Plate_5_sc_feature_selected.parquet").resolve(strict=True)
+
+plate3df = pd.read_parquet(plate3df_path)
+plate3pdf = pd.read_parquet(plate3pdf_path)
+plate5df = pd.read_parquet(plate5df_path)
+
+
+# ### Output paths
+
+# In[4]:
+
+
+fig_path3 = pathlib.Path("plate_3_figures")
+fig_path3p = pathlib.Path("plate_3_prime_figures")
+fig_path5 = pathlib.Path("plate_5_figures")
+
+fig_path3.mkdir(parents=True, exist_ok=True)
+fig_path3p.mkdir(parents=True, exist_ok=True)
+fig_path5.mkdir(parents=True, exist_ok=True)
+
+# Map plate names to path for figure generation
+plate_name_to_path_map = {
+    "plate_3": fig_path3,
+    "plate_3p": fig_path3p,
+    "plate_5": fig_path5
+}
+
+
+# ## Processing Functions
+
+# In[5]:
+
+
+cell_count_col = "Metadata_number_of_singlecells"
+
+def agg_cell_count_to_site(_platedf):
+    """
+    Determine cell counts at the site level.
+
+    Parameters
+    ----------
+    _platedf: Pandas Dataframe
+        Single-cell morphology data for a single plate
+
+    Returns
+    -------
+    site_cell_countdf: Pandas Dataframe
+       Cell counts across sites with genotype and edge well columns
+    """
+
+    site_cell_countdf = (
+        _platedf.groupby(["Metadata_Well", "Metadata_Site"])
+        .agg(
+        {
+            cell_count_col: "count",
+            "Metadata_genotype": get_first_element,
+            "Metadata_Edge_Well": get_first_element
+        })
+    )
+
+    site_cell_countdf.reset_index(inplace=True)
+
+    return site_cell_countdf
+
+
+# In[6]:
+
+
+def agg_cell_count_genotype_edge_well(_df):
+    """
+    Determine cell counts for each unique (genotype, edge well) pair.
+
+    Parameters
+    ----------
+    _df: Pandas Dataframe
+        Single-cell morphology data for a single plate
+
+    Returns
+    -------
+    _df: Pandas Dataframe
+        Number of single-cells for each unique (genotype, edge well) pair.
+    """
+
+    _df = _df.groupby(["Metadata_genotype", "Metadata_Edge_Well"]).agg(
+        {
+        "Metadata_number_of_singlecells": "count",
+        }).reset_index()
+
+    return _df
+
+
+# In[7]:
+
+
+def combine_edge_well_data(_genotype_cell_countdf):
+    """
+    Parameters
```

Comment:
Recommend including a short snippet describing the function to keep the same format as the above functions.

Reply:
I thought about this, but decided to not include a description to keep these comments DRY, since I would be essentially echoing the `Return` comment in the description